Composer/packages/client/src/components/TriggerCreationModal/TriggerOptionTree.ts

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { TriggerUISchema } from '@bfc/extension-client';
+
+import { TriggerUIOption } from './schema/TriggerOption';
+
+export class TriggerOptionLeafNode {
+  label: string;
+  $kind: string;
+  parent: TriggerOptionGroupNode | null = null;
+
+  constructor(label: string, $kind: string) {
+    this.label = label;
+    this.$kind = $kind;
+  }
+}
+
+export class TriggerOptionGroupNode {
+  label: string;
+  /** Title of a dropdown. 'Which activity type?' */
+  prompt?: string;
+  /** Placeholder of a dropdown input. 'Select an activity type' */
+  placeholder?: string;
+  children: (TriggerOptionLeafNode | TriggerOptionGroupNode)[] = [];
+  parent: TriggerOptionGroupNode | null = null;
+
+  constructor(label: string, prompt?: string, placeholder?: string) {
+    this.label = label;
+    this.prompt = prompt;
+    this.placeholder = placeholder;
+  }
+}
```

Comment:
What is the value of using classes instead of typed object literals?

Reply:
One of the reason is I need to rely on the `instanceof` operator to determine type of a tree node.
Here is an example:
<img src="https://user-images.githubusercontent.com/8528761/99505519-e440b080-29bb-11eb-8ead-5e209cb9c8d4.png" width="500" />